mono/utils/strenc.h

```diff
@@ -14,7 +14,7 @@
 #include <glib.h>
 #include <mono/utils/mono-publib.h>
 
-extern MONO_API gunichar2 *mono_unicode_from_external (const gchar *in, gsize *bytes);
+extern MONO_API gunichar2 *mono_unicode_from_external (const gchar *in, guint32 *bytes);
 extern MONO_API gchar *mono_utf8_from_external (const gchar *in);
```

Comment:
This is public api, so it cannot be changed.

Reply:
What options do we have then? I guess I could leave it as `gsize`, then initialize the value inside `mono_unicode_from_external`.

Reply:
Yes, unfortunately it needs to use a different local with a different type.

Reply:
Ok, I'll make that change then. Thanks!

Reply:
After some more investigation, the change to the implementation of `mono_unicode_from_external` is really all that we need to correct the problem we see. So I'll close this PR and open a new one with a smaller change.